Realm/Realm/Handles/CollectionHandleBase.cs

```diff
@@ -55,20 +55,15 @@ public ResultsHandle Snapshot()
 
         public abstract void Clear();
 
-        protected RealmValue ToRealmValue(PrimitiveValue primitive, RealmObjectBase.Metadata metadata, Realm realm)
+        protected RealmValue ToRealmValue(PrimitiveValue primitive, Realm realm)
         {
             if (primitive.Type != RealmValueType.Object)
             {
                 return new RealmValue(primitive);
             }
 
-            var objectHandle = primitive.AsObject(Root);
-            if (metadata == null)
-            {
-                throw new NotImplementedException("Mixed objects are not supported yet.");
-            }
-
-            return new RealmValue(realm.MakeObject(metadata, objectHandle));
+            (var objectHandle, var tableKey) = primitive.AsObject(Root);
```

Comment:
[extreme nit] I prefer the single var syntax for tuple deconstruction: `var (objectHandle, tableKey) = ...` 😄